src/coreclr/src/jit/morph.cpp

```diff
@@ -14767,10 +14769,18 @@ bool Compiler::fgFoldConditional(BasicBlock* block)
             noway_assert(cond->gtOper == GT_CNS_INT);
             noway_assert((block->bbNext->countOfInEdges() > 0) && (block->bbJumpDest->countOfInEdges() > 0));
 
-            /* remove the statement from bbTreelist - No need to update
-             * the reference counts since there are no lcl vars */
-            fgRemoveStmt(block, lastStmt);
-
+            if (condTree != cond)
+            {
+                // Preserve the potentially side effecting part of the comma
+                assert(condTree->OperIs(GT_COMMA));
+                lastStmt->SetRootNode(condTree->AsOp()->gtOp1);
```

Comment:
Can condTree->AsOp()->gtOp2 be a GT_COMMA tree too?

Reply:
Yes, You could have multiple GT_COMMA nodes and need to collect all of the side effects not just the first one.

Reply:
The tree is a JTRUE, so there is no op2.

Reply:
So, we should set the root node to `condTree`, not `condTree->op1`.

Reply:
`condTree` is a `GT_COMMA` so it has op1 and op2.

Reply:
Yeah I wasn't clear, I meant the root tree is JTRUE, so we can just replace it with the COMMA.

Reply:
I still don't understand. If `condTree` is `COMMA(x, COMMA(y, z))` won't we lose the size effects from `y`?

Reply:
We'd turn `JTRUE(COMMA(x, COMMA(y, 1)))` into `COMMA(x, COMMA(y, 1))`

Reply:
@AndyAyersMS Sorry, I was confused yesterday because I saw this message

>> The tree is a JTRUE, so there is no op2.

but didn't notice this one

>> So, we should set the root node to condTree, not condTree->op1.

Your suggestion makes sense.